src/bp-core/bp-core-functions.php

```diff
@@ -660,12 +660,14 @@ function bp_core_get_directory_pages() {
  *
  * @since BuddyPress 1.7.0
  *
- * @param array  $components Components to create pages for.
- * @param string $existing   'delete' if you want to delete existing page mappings
- *                           and replace with new ones. Otherwise existing page mappings
- *                           are kept, and the gaps filled in with new pages. Default: 'keep'.
+ * @param array  $components   Components to create pages for.
+ * @param string $existing     'delete' if you want to delete existing page mappings
+ *                             and replace with new ones. Otherwise existing page mappings
+ *                             are kept, and the gaps filled in with new pages. Default: 'keep'.
+ * @param bool   $map_register Whether or not mapping the registration pages.
+ *                             Default: true
  */
-function bp_core_add_page_mappings( $components, $existing = 'keep' ) {
+function bp_core_add_page_mappings( $components, $existing = 'keep', $map_register = true ) {
```

Comment:
@surajkrsingh I think the `map_register` name is not correct this flag is not only for registration we checking for privacy policy also I think we choose an appropriate name something like `page_mapping`

Reply:
@dipskakadiya
With this param,  we are just checking the registration pages.
![image](https://user-images.githubusercontent.com/19591301/143992756-ccf082ac-a743-4b5f-bdce-024f90918205.png)

So, that it's is `$map_register` 
Let me know if still need to rename the param.